unit_tests/sources/declarative/decoders/test_composite_decoder.py

```diff
@@ -202,6 +204,30 @@ def test_composite_raw_decoder_csv_parser_values(requests_mock, encoding: str, d
     assert parsed_records == expected_data
 
 
+class TestServer(BaseHTTPRequestHandler):
+    def do_GET(self) -> None:
+        self.send_response(200)
+        self.end_headers()
+        self.wfile.write(bytes("col1,col2\nval1,val2", "utf-8"))
+
+
+def test_composite_raw_decoder_csv_parser_without_mocked_response():
+    """
+    This test reproduce a `ValueError: I/O operation on closed file` error we had with CSV parsing. We could not catch this with other tests because the closing of the mocked response from requests_mock was not the same as the one in requests.
+    """
+    # start server
+    httpd = HTTPServer(("localhost", 8080), TestServer)
```

Comment:
We should probably have a better way to find a port. Is there a port finder in Python?

Reply:
I guess we can use `socket` package to do it

Reply:
Let's go with this for now and if we see issues, we can migrate to socket directly